lib/components/form/connect-location-field.js

```diff
@@ -0,0 +1,51 @@
+import { connect } from 'react-redux'
+
+import * as locationActions from '../../actions/location'
+import * as apiActions from '../../actions/api'
+import { getActiveSearch, getShowUserSettings } from '../../util/state'
+
+/**
+ * This higher-order component connects the target (styled) LocationField to the
+ * redux store.
+ * @param StyledLocationField The input LocationField component to connect.
+ * @param options Optional object with the following optional props:
+ *                - actions: a list of actions to include in mapDispatchToProps
+ *                - includeLocation: whether to derive the location prop from
+ *                  the active query
+ * @returns The connected component.
+ */
+export default function connectLocationField (StyledLocationField, options = {}) {
+  // By default, set actions to empty list and do not include location.
+  const {actions = [], includeLocation = false} = options
+  const mapStateToProps = (state, ownProps) => {
+    const { config, currentQuery, location, transitIndex, user } = state.otp
+    const { currentPosition, nearbyStops, sessionSearches } = location
+    const activeSearch = getActiveSearch(state.otp)
+    const query = activeSearch ? activeSearch.query : currentQuery
+
+    const stateToProps = {
+      currentPosition,
+      geocoderConfig: config.geocoder,
+      nearbyStops,
+      sessionSearches,
+      showUserSettings: getShowUserSettings(state.otp),
+      stopsIndex: transitIndex.stops,
+      userLocationsAndRecentPlaces: [...user.locations, ...user.recentPlaces]
+    }
+    // Leave stateToProps.location completely unset unless includeLocation is true.
+    if (includeLocation) {
```

Comment:
@landonreed It is still necessary to leave the `location` prop unset, otherwise the `IntermediatePlaceField` and `FavoriteLocation` components will use the fixed `null` value as location.

Reply:
Ah gotcha, thanks for the info. Could you include that in the comment above?

Reply:
Yes, updated in https://github.com/opentripplanner/otp-react-redux/pull/284/commits/cae2ba746809b92e8b103378e5752c2c385ec492.